sip/sip-002-smart-contract-language.md

```diff
@@ -83,6 +83,18 @@ there may be some database effects which are materialized from
 Unlike functions created by `define-public`, which may only return booleans,
 functions created with `define-read-only` may return any type.
 
+## List Operations
+
+* Lists may be multi-dimensional. However, note that runtime admission checks
+  on typed function-parameters and data-map functions like `set-entry!` will
+  be charged based on the _maximal_ size of the multi-dimensional list.
+* `filter` `map` and `fold` functions may only be called with user-defined functions
```

Comment:
From the code, it looks like you can use either a user-defined function or a native simple function (like `+`)?

Reply:
Ah, yes. I will correct in the md file, I got a little ahead of myself there.

A later patch will address the typechecking of what the typechecker currently treats as "special functions" (e.g., something like `eq?`), which should still be easily type-checked.

The only functions which won't be allowed to fold/map/filter will be functions that have non-evaluated/name arguments, e.g., the datamap functions which require a map name argument, contract-call which requires the function-name argument, etc., and of course, functions that don't have matching input/output type signatures.

Reply:
Added a note for using "simple native functions".

Reply:
Thanks!